src/main/java/org/alliancegenome/curation_api/jobs/BulkLoadJobExecutor.java

```diff
@@ -82,115 +116,142 @@ public void process(BulkLoadFile bulkLoadFile) throws Exception {
-				throw new Exception("Ontolgy Load: " + bulkLoadFile.getBulkLoad().getName() + " for OT: " + ontologyType + " not implemented");
+
+			switch (ontologyType) {
+				case ZECO:
+					config.setLoadOnlyIRIPrefix("ZECO");
+					service = zecoTermService;
+					processTerms(bulkLoadFile, zecoTermService, config);
+					break;
+				case EMAPA:
+					config.getAltNameSpaces().add("anatomical_structure");
+					service = emapaTermService;
+					processTerms(bulkLoadFile, emapaTermService, config);
+					break;
+				case GO:
+					config.getAltNameSpaces().add("biological_process");
+					config.getAltNameSpaces().add("molecular_function");
+					config.getAltNameSpaces().add("cellular_component");
+					service = goTermService;
+					processTerms(bulkLoadFile, goTermService, config);
+					break;
+				case SO:
+					processTerms(bulkLoadFile, soTermService, config);
+					break;
+				case XCO:
+					processTerms(bulkLoadFile, xcoTermService, config);
+					break;
+				case ECO:
+					processTerms(bulkLoadFile, ecoTermService, config);
+					ecoTermService.updateAbbreviations();
+					break;
+				case CHEBI:
+					processTerms(bulkLoadFile, chebiTermService, config);
+					break;
+				case ZFA:
+					config.getAltNameSpaces().add("zebrafish_anatomy");
+					processTerms(bulkLoadFile, zfaTermService, config);
+					break;
+				case DO:
+					processTerms(bulkLoadFile, doTermService, config);
+					break;
+				case MP:
+					config.setLoadOnlyIRIPrefix("MP");
+					processTerms(bulkLoadFile, mpTermService, config);
+					break;
+				case MA:
+					processTerms(bulkLoadFile, maTermService, config);
+					break;
+				case WBBT:
+					processTerms(bulkLoadFile, wbbtTermService, config);
+					break;
+				case DAO:
+					config.setLoadOnlyIRIPrefix("FBbt");
+					processTerms(bulkLoadFile, daoTermService, config);
+					break;
+				case WBLS:
+					processTerms(bulkLoadFile, wblsTermService, config);
+					break;
+				case FBDV:
+					processTerms(bulkLoadFile, fbdvTermService, config);
+					break;
+				case MMUSDV:
+					config.getAltNameSpaces().add("mouse_developmental_stage");
+					config.getAltNameSpaces().add("mouse_stages_ontology");
+					processTerms(bulkLoadFile, mmusdvTermService, config);
+					break;
+				case ZFS:
+					processTerms(bulkLoadFile, zfsTermService, config);
+					break;
+				case XBA_XBS:
+					config.getAltNameSpaces().add("xenopus_anatomy");
+					config.getAltNameSpaces().add("xenopus_anatomy_in_vitro");
+					processTerms(bulkLoadFile, OntologyBulkLoadType.XBA, xbaTermService, config);
+					GenericOntologyLoadConfig config2 = new GenericOntologyLoadConfig();
+					config2.getAltNameSpaces().add("xenopus_developmental_stage");
+					processTerms(bulkLoadFile, OntologyBulkLoadType.XBS, xbsTermService, config2);
+					break;
+				case XPO:
+					config.setLoadOnlyIRIPrefix("XPO");
+					processTerms(bulkLoadFile, xpoTermService, config);
+					break;
+				case ATP:
+					processTerms(bulkLoadFile, atpTermService, config);
```

Comment:
@cmpich you will need to do a `config.setLoadOnlyIRIPrefix("ATP");` here also.

Reply:
@cmpich the config needs to happen before the process happens.

Reply:
Sure.